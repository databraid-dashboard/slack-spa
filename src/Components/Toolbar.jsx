// @flow
/* eslint-disable */
import React, { Component } from 'react';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';
import { Dropdown, Image, Menu } from 'semantic-ui-react';
import { selectChannel, fetchChannels, fetchScoreForChannel } from '../Actions/index';
import convertScoreToColorAndEmoji from '../Utils/';
import frustrated from '../images/emojis/frustrated.jpg';
import happy from '../images/emojis/happy.jpg';
import neutral from '../images/emojis/neutral.jpg';
import sad from '../images/emojis/sad.jpg';
import slack from '../images/slackIcon.png';
import smile from '../images/emojis/smile.jpg';
import '../index.css';
import type { ChannelData } from '../FlowTypes/';
import injectWidgetId from '../Utils/utils';
import type { Dispatch, OwnProps, State } from '../FlowTypes/';

const sentiments = {
  frustrated,
  sad,
  neutral,
  smile,
  happy,
};

export class Toolbar extends Component {
  componentWillMount() {
    const channels = Object.keys(this.props.channelData);
    if (channels.length === 0) {
      this.props.fetchChannels();
    }
  }

  props: {
    score: mixed,
    channelData: ChannelData,
    fetchChannels: Function,
    selectChannel: mixed,
    selectedChannel: mixed,
    fetchScoreForChannel: Function,
  };

  render() {
    const { score, selectedChannel, channelData, selectChannel, fetchScoreForChannel } = this.props;

    if (!score) {
      if (selectedChannel) {
        fetchScoreForChannel(selectedChannel);
      }
    }

    const currentSentiment = convertScoreToColorAndEmoji(score).emoji;
    const computedColor = convertScoreToColorAndEmoji(score).color;
    const menuClasses = `ui ${computedColor.slice(6)} inverted menu`;

    return (
      <Menu size="small" inverted className={menuClasses}>
        <Menu.Item className="ui button">
          <Image avatar src={slack} />
        </Menu.Item>
        <Dropdown item text={selectedChannel || 'Select a channel'}>
          <Dropdown.Menu>
            {Object.keys(channelData).map(channel => (
              <Dropdown.Item
                key={channel}
                onClick={() => selectChannel(channel)}
                selected={channel === selectedChannel}
              >
                {channel}
              </Dropdown.Item>
            ))}
          </Dropdown.Menu>
        </Dropdown>
        <Menu.Menu position="right">
          <Menu.Item className="ui button">
            <Image avatar src={sentiments[currentSentiment]} />
          </Menu.Item>
        </Menu.Menu>
      </Menu>
    );
  }
}

export const mapStateToProps = (state: State, ownProps: OwnProps) => {
  const id = ownProps.widgetId;
  const selectedChannel = state.widgets.byId[id].selectedChannel;
  const score = state.widgets.byId[id].scoreData[selectedChannel];
  const channelData = state.widgets.byId[id].channelData;

  return {
    channelData,
    score,
    selectedChannel,
  };
};

export const mapDispatchToProps = (dispatch: Dispatch) =>
  bindActionCreators(
    {
      selectChannel,
      fetchChannels,
      fetchScoreForChannel,
    },
    dispatch,
  );

export default injectWidgetId(connect(mapStateToProps, mapDispatchToProps)(Toolbar));
