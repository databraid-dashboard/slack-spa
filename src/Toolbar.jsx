// @flow

import { Dropdown, Image, Menu } from 'semantic-ui-react';
import React from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import slack from './images/slackIcon.png';
import frustrated from './images/emojis/frustrated.jpg';
import sad from './images/emojis/sad.jpg';
import neutral from './images/emojis/neutral.jpg';
import happy from './images/emojis/happy.jpg';
import smile from './images/emojis/smile.jpg';
import convertScoreToColorAndEmoji from './convertScoreToColorAndEmoji';

import './index.css';
// import store from './store';
import Actions from './Actions';
import {selectChannel, fetchChannels} from './Actions';

const sentiments =
  {
    frustrated,
    sad,
    neutral,
    smile,
    happy,
  };


export class Toolbar extends React.Component {
  componentWillMount() {
    const channels = Object.keys(this.props.channelData);
    if (channels.length === 0) {
      this.props.fetchChannels()
    }
  }

  props: {
    //color: string,
    isShowingScores: boolean,
    score: mixed,
  };

  render() {
    const { score, selectedChannel, channelData, selectChannel } = this.props;
    console.log(score);
    //const { selectedChannel } = store.getState();
    const currentSentiment = convertScoreToColorAndEmoji(score).emoji;
    const computedColor = convertScoreToColorAndEmoji(score).color;
    const menuClasses = `ui ${computedColor} inverted menu`;

    return (
      <Menu size="small" className={menuClasses}>
        <Menu.Item className="ui button">
          <Image avatar src={slack} />
        </Menu.Item>
        <Dropdown item text={selectedChannel || 'Select a channel'}>
          <Dropdown.Menu>
            {Object.keys(channelData).map(
              channel =>
                (<Dropdown.Item
                  key={channel}
                  onClick={() => selectChannel(channel)}
                  selected={channel === selectedChannel}
                >
                  {channel}
                </Dropdown.Item>),
            )}
          </Dropdown.Menu>
        </Dropdown>
        <Menu.Menu position="right">
          <Menu.Item className="ui button">
            <Image
              avatar
              src={sentiments[currentSentiment]}
            />
          </Menu.Item>
        </Menu.Menu>
      </Menu>
    );
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    selectedChannel: state.selectedChannel,
    score: state.scoreData[state.selectedChannel],
    channelData: state.channelData,

  }
};
const mapDispatchToProps = (dispatch) =>{

  return bindActionCreators({selectChannel, fetchChannels}, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(Toolbar)
