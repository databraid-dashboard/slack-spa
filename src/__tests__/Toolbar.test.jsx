import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';
import Toolbar from '../Components/Toolbar';

describe('Renders toolbar with default props', () => {
  it('should render', () => {
    const toolbar = shallow(<Toolbar />);
    expect(toJson(toolbar)).toMatchSnapshot();
  });
});
