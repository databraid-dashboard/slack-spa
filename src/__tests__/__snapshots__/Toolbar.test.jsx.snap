// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders toolbar with default props should render 1`] = `<Connect(Toolbar) />`;
